src/main/java/com/exactpro/th2/common/schema/message/FilterFunction.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.exactpro.th2.common.schema.message;
+
+import com.exactpro.th2.common.schema.filter.strategy.FilterStrategy;
+import com.exactpro.th2.common.schema.filter.strategy.impl.DefaultFilterStrategy;
+import com.exactpro.th2.common.schema.message.configuration.RouterFilter;
+import com.google.protobuf.Message;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+public interface FilterFunction extends BiFunction<Message, List<? extends RouterFilter>, Boolean> {
```

Comment:
```suggestion
public interface <T extends Message> FilterFunction extends BiFunction<T, List<? extends RouterFilter>, Boolean> {
```

Reply:
I am facing a problem when trying to add generic to `FilterFunction`. In `AbstractMessageQueue` we can use `<T>`, but for the default filter function I used `? extends Message` and when we what to set one to a field we can not cast `? extends Message` to `T`, which `T extends Message`. Which generic we should set for the default filter function?

Reply:
Let's transform the DefaultFilterStrategy class to an AbstractFilterStrategy then move logic from FieldExtractionStrategys to implementations of the FilterStrategy interface